meta-balena-common/recipes-support/balena-config-vars/balena-config-vars/balena-config-vars

```diff
@@ -117,18 +111,34 @@ else
         if [ "$RESULT" = "4" ]; then
             echo "[ERROR] $0: Failed to parse '$CONFIG_PATH'."
             exit 1
-        else
-            if [ -n "${BALENA_CONFIG_VARS_CACHE}" ]; then
-                tmpfile=$(mktemp)
-                echo "${CONFIG_PARAMS}" | sed -e 's/^[[:space:]]*//' > "${tmpfile}"
-                chmod a+rx "${tmpfile}"
-                mv "${tmpfile}" "${BALENA_CONFIG_VARS_CACHE}"
-            fi
-            eval "$CONFIG_PARAMS"
         fi
+        test -n "${CONFIG_PARAMS}"
     else
         echo "[WARNING] $0 : '$CONFIG_PATH' not found."
     fi
+}
+
+if [ "${USE_CACHE}" -eq "1" ] && [ -n "${BALENA_CONFIG_VARS_CACHE}" ] && [ -f "${BALENA_CONFIG_VARS_CACHE}" ]; then
+        . "${BALENA_CONFIG_VARS_CACHE}"
+else
+    [ -n "${BALENA_CONFIG_VARS_CACHE}" ] && [ -f "${BALENA_CONFIG_VARS_CACHE}" ] && rm -f "${BALENA_CONFIG_VARS_CACHE}"
+
+    # If config.json provides redefinitions for our vars let us rewrite their
+    # runtime value
+    if ! read_config; then
```

Comment:
@floion the return code is then checked here - that's the intention anyway. I am about to build and test it.